bindata/network/ovn-kubernetes/ovnkube-master.yaml

```diff
@@ -329,10 +364,34 @@ spec:
             fi
           fi
 
+          function watch_core_dns_ep () {
+            set +ex
+
+            while true; do
+              is_dns_up=$(kubectl get ep dns-default -n openshift-dns -o jsonpath='{.subsets[0].addresses[0].ip}' 2>/dev/null)
+              if [[ -n "${is_dns_up}" ]]; then
+                break
+              fi
+              echo "[$(date +%Y-%m-%dT%H:%M:%SZ)][watch_core_dns_ep] waiting for CoreDNS to start..."
+              sleep 5
+            done
+
+            echo "[$(date +%Y-%m-%dT%H:%M:%SZ)][watch_core_dns_ep] CoreDNS is up, replacing inherited nameserver entry from host with CoreDNS service IP in /etc/resolv.conf"
+            core_dns_svc_ip=$(kubectl get svc dns-default -n openshift-dns -o jsonpath='{.spec.clusterIP}' 2>/dev/null)
+
+            cp /etc/resolv.conf resolv.tmp
+            sed -i "s/nameserver.*/nameserver $core_dns_svc_ip/g" resolv.tmp
+            cp -f resolv.tmp /etc/resolv.conf
+            rm resolv.tmp
+          }
+
+          watch_core_dns_ep &
+
           # start nbctl daemon for caching
           export OVN_NB_DAEMON=$(ovn-nbctl --pidfile=/run/openvswitch/ovnk-nbctl.pid \
             --detach \
             -p /ovn-cert/tls.key -c /ovn-cert/tls.crt -C /ovn-ca/ca-bundle.crt \
+            --no-leader-only \
```

Comment:
The man page makes it sound like this is a bad idea... ("means that for read-only transactions it can report and act on stale data")

Reply:
Well, unfortunately that is a hard constraint for integrating all of this. Without `--no-leader-only` the `ovn-nbctl` commands risk talking to a follower and fail. Maybe we could have a discussion during the OVN weekly next week concerning this?

Reply:
OK, so based on my reading the "Understanding Cluster Consistency" section of `ovsdb(7)`... it seems like as long as we're consistently talking to the same follower then we'll be fine; we might be slightly behind the leader at any point in time, but we'll eventually catch up. It would only be a problem if we talked to two different masters back-to-back when they were out of sync. (eg, if we create an object on master A and then query master B about its attributes). And it seems like `ovn-nbctl` daemon mode ought to mostly prevent that from happening, as long as the masters are stable. So maybe we're ok?

Reply:
Hm... well, actually, etcd will not be synchronized with the ovn databases... So if there are situations where we assume that you can do:
- node 1 writes X=Y to ovn
- node 1 writes A=B to etcd
- node 2 sees that A=B has been written to etcd
- node 2 reads X from ovn and gets Y

then that might break now; node 2 might get the previous value of X instead

Reply:
Crap, you're right. I need to think about this but I wonder if we avoid the problem if ovnkube-master daemon only talks to the db deployed on his node. There will (hrmmm, should...) only be one ovnkube-master leader at any given point in time. ovnkube-node performs transactions using ovn-nbctl, but I'm not mistaken, those are only node specific and do not involve any transaction caused by changing k8s state.

Reply:
So, I think we should be alright if we remove `--db` when launching the daemon. `ovnkube-master` will thus only execute commands against the db member running on the same node (whether it's the leader or not). Master is the only one acting on k8s state. Well, we have multiple `AddServiceHandler` / `AddEndpointsHandler` in the code, but they all do different things and _should_ not overlap and act on the same OVN object - @dcbw, I guess your input is needed on this.

I will update the PR with this for now.